File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;
class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        
        Arrays.sort(score);

        int min = 10;
        int index = 0;

        for(int i = score.length - 1; i >= 0; i--) {
            min = min > score[i] ? score[i] : min;
            index++;
            
            if(index == m){
                index = 0;
                answer += min * m;
                min = 10;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return (*(int*)b - *(int*)a);
}

int solution(int k, int m, int score[], size_t score_len) {
    int answer = 0;
    int box = score_len / m;
    
    qsort(score, score_len, sizeof(int), compare);
    
    for(int i = 0; i < box; i ++){
        answer += score[(i + 1) * m - 1] * m;
    }
    
    
    return answer;
}